### PR TITLE
refactor(arn): fine tune arn and resources id for unknown values

### PR DESCRIPTION
**Pull request by pedrooot**
### Description

This pull request includes significant updates to the `prowler` codebase, primarily focusing on enhancing the handling of ARNs (Amazon Resource Names) by including resource types. The changes span across multiple files and involve modifications to functions, test cases, and exception handling.

Enhancements to ARN handling:

* [`prowler/providers/aws/lib/service/service.py`](diffhunk://#diff-2a80643e99a3dd3dd5dd072b20bdd17d09379a0d5fbb28c54de827dffb314990L105-R117): Updated the `get_unknown_arn` method to accept an optional `resource_type` parameter, allowing more specific ARNs to be generated.

Updates to service-specific ARN handling:

* [`prowler/providers/aws/services/accessanalyzer/accessanalyzer_service.py`](diffhunk://#diff-f02f4768cee477f4d1d9d1f5f3458d8b290c2a716a048866538a08831ae7cb7bL46-R49): Modified the `_list_analyzers` method to include `resource_type="analyzer"` when calling `get_unknown_arn`.
* [`prowler/providers/aws/services/guardduty/guardduty_service.py`](diffhunk://#diff-31217e83ad2d760018402f9d31202f267ba46ad12c147b91724b8c6ff68fd6a5L45-R48): Updated the `_list_detectors` method to include `resource_type="detector"` when calling `get_unknown_arn`.
* [`prowler/providers/aws/services/securityhub/securityhub_service.py`](diffhunk://#diff-16776ee6aad977fc3c6f7ab2aa22220168fb6a8d909dc5301023fa78854e637eL30-R33): Adjusted the `_describe_hub` method to include `resource_type="hub"` in two instances when calling `get_unknown_arn`. [[1]](diffhunk://#diff-16776ee6aad977fc3c6f7ab2aa22220168fb6a8d909dc5301023fa78854e637eL30-R33) [[2]](diffhunk://#diff-16776ee6aad977fc3c6f7ab2aa22220168fb6a8d909dc5301023fa78854e637eL76-R81)

Test case updates:

* [`tests/providers/aws/lib/service/service_test.py`](diffhunk://#diff-c38a008d5b655049a59118ab2503f17e9913dc25b2ee9f9812e64a64148195e4R124-R154): Added new test cases to verify the `get_unknown_arn` method with the `resource_type` parameter set.
* [`tests/providers/aws/services/accessanalyzer/accessanalyzer_enabled/accessanalyzer_enabled_test.py`](diffhunk://#diff-1a144e0bd82864df98cb8f8468d920f5ce33433f1a0047984478b2aac7df83c7L129-R134): Updated the `test_two_analyzers` method to reflect changes in ARN generation for analyzers. [[1]](diffhunk://#diff-1a144e0bd82864df98cb8f8468d920f5ce33433f1a0047984478b2aac7df83c7L129-R134) [[2]](diffhunk://#diff-1a144e0bd82864df98cb8f8468d920f5ce33433f1a0047984478b2aac7df83c7L156-R156) [[3]](diffhunk://#diff-1a144e0bd82864df98cb8f8468d920f5ce33433f1a0047984478b2aac7df83c7L173-R177)
* [`tests/providers/aws/services/securityhub/securityhub_enabled/securityhub_enabled_test.py`](diffhunk://#diff-dff77a6e18517a3712fd9b365c102469c2aa8a6e417d35abd2c38378b93b7d78L14-R19): Modified the `test_securityhub_hub_inactive` method to accommodate the new `resource_type` parameter in ARNs for security hubs. [[1]](diffhunk://#diff-dff77a6e18517a3712fd9b365c102469c2aa8a6e417d35abd2c38378b93b7d78L14-R19) [[2]](diffhunk://#diff-dff77a6e18517a3712fd9b365c102469c2aa8a6e417d35abd2c38378b93b7d78L32-R32) [[3]](diffhunk://#diff-dff77a6e18517a3712fd9b365c102469c2aa8a6e417d35abd2c38378b93b7d78L44-R47)

These changes improve the specificity and accuracy of ARNs generated across various AWS services, enhancing the overall robustness of the `prowler` tool.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [ ] Review if the code is being covered by tests.
- [ ] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [ ] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/developer-guide/checks.md [hunk 1]
```diff
@@ -160,16 +160,21 @@ else:
...
+            - We create a valid ARN with the resource_id part as the resource audited. Examples:
+                - Bedrock -> `arn:<partition>:bedrock:<region>:<account-id>:model-invocation-logging`
+                - DirectConnect -> `arn:<partition>:directconnect:<region>:<account-id>:dxcon`
+        - If there is no real resource to audit we do the following:
+            - resource_id -> `resource_type/unknown`
+            - resource_arn -> `arn:<partition>:<service>:<region>:<account-id>:<resource_type>/unknown`
+            This replaces having the Account ID and the Account ID ARN as the values if the above happens. Examples:
+                - AWS Security Hub -> arn:<partition>:security-hub:<region>:<account-id>:hub/unknown
+                - Access Analyzer -> arn:<partition>:access-analyzer:<region>:<account-id>:analyzer/unknown
+                - GuardDuty -> arn:<partition>:guardduty:<region>:<account-id>:detector/unknown
```
Comment (jfagoagas) [inline docs/developer-guide/checks.md]:
```suggestion
    - Resouce ID and resource ARN:
        - If the resource audited is the AWS account:
            - `resource_id` -> AWS Account Number
            - `resource_arn` -> AWS Account Root ARN
        - If we can’t get the ARN from the resource audited, we create a valid ARN with the `resource_id` part as the resource audited. Examples:
                - Bedrock:
                  - `resource_id` -> `model-invocation-logging`
                  - `resource_arn` -> `arn:<partition>:bedrock:<region>:<account-id>:model-invocation-logging`
                - DirectConnect: 
                  - `resource_id` -> `dxcon`
                  - `resource_arn` -> `arn:<partition>:directconnect:<region>:<account-id>:dxcon`
        - If there is no real resource to audit:
            - resource_id -> `resource_type/unknown`
            - resource_arn -> `arn:<partition>:<service>:<region>:<account-id>:<resource_type>/unknown`
            This replaces having the Account ID and the Account ID ARN as the values if the above happens. Examples:
                - AWS Security Hub -> arn:<partition>:security-hub:<region>:<account-id>:hub/unknown
                - Access Analyzer -> arn:<partition>:access-analyzer:<region>:<account-id>:analyzer/unknown
                - GuardDuty -> arn:<partition>:guardduty:<region>:<account-id>:detector/unknown
```

**Review by jfagoagas — COMMENTED** [review]
Just one comment about the docs.

File: docs/developer-guide/checks.md (hunk 1 above)
Comment (pedrooot) [reply docs/developer-guide/checks.md]: Done

**Review by jfagoagas — DISMISSED** [review]
💯

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**